src/metabase/actions.clj

```diff
@@ -136,42 +114,38 @@
           (swap! *misc-value-cache* assoc unique-key value))
         value)))
 
+(defn check-if-actions-enabled!
+  "Throws an appropriate error if actions are unsupported or disabled, otherwise returns nil."
+  [{db-settings :settings db-id :id driver :engine db-name :name :as db}]
+  (when-not (driver/database-supports? driver :actions db)
+    (throw (ex-info (i18n/tru "{0} Database {1} does not support actions."
+                              (u/qualified-name driver)
+                              (format "%d %s" db-id (pr-str db-name)))
+                    {:status-code 400, :database-id db-id})))
+
+  (binding [setting/*database-local-values* db-settings]
+    (when-not (database-enable-actions)
+      (throw (ex-info (i18n/tru "Actions are not enabled.")
+                      {:status-code 400, :database-id db-id}))))
+
+  nil)
```

Comment:
Nice! I would call have called this `check-actions-enabled` though, to be consistent with other `check-**-enabled` function names.